samples/xapi-server/src/main/java/dev/learning/xapi/samples/xapiserver/StateEntity.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2016-2025 Berry Cloud Ltd. All rights reserved.
+ */
+
+package dev.learning.xapi.samples.xapiserver;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.IdClass;
+import jakarta.persistence.Lob;
+import java.io.Serializable;
+import java.util.UUID;
+
+/**
+ * StateEntity.
+ *
+ * @author Thomas Turrell-Croft
+ */
+@Entity
+@IdClass(StateEntity.StateId.class)
+public class StateEntity {
+
+  @Id
+  @Column(length = 2048)
+  private String activityId;
+
+  @Id
+  @Column(length = 2048)
+  private String agentJson;
+
+  @Id
+  private String stateId;
+
+  private UUID registration;
```

Comment:
registration can be part of the composite id since we treat no registration as a 0 based uuid